flashlight/fl/tensor/backend/jit/eval/Evaluator.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <unordered_map>
+
+#include "flashlight/fl/tensor/TensorBackend.h"
+#include "flashlight/fl/tensor/TensorBase.h"
+#include "flashlight/fl/tensor/backend/jit/ir/BinaryNode.h"
+#include "flashlight/fl/tensor/backend/jit/ir/CustomNode.h"
+#include "flashlight/fl/tensor/backend/jit/ir/ScalarNode.h"
+
+namespace fl {
+
+/**
+ * A JIT tree evaluator. It dispatches to another Tensor Backend for carrying
+ * out the computation represented by the JIT tree.
+ */
+class Evaluator {
```

Comment:
Random idea, but is there any reason we might want to subclass this?

Reply:
Can't really think of one yet -- I've had thoughts on supporting backend-specific Opaque node (from optimization, autograd, etc.), which requires the backend to inject evaluation logic for those nodes, which can be implemented either by inheritance or some TensorExtension-style thing.